### PR TITLE
Adds a defibrillator upgrade in R&D

### PR DESCRIPTION
**Pull request by XDTM**
:cl: XDTM
add: Hyper-Conductor Upgrade for Defibrillators : Printed in R&D, it'll let you shock even the most decayed corpses back to life! It will, however, increase the amount of energy spent when used.
/:cl:

It's used like a KA modkit, which also means that science can't benefit from this unless they outright steal a defib.
Benefits:
- Can be used on corpses over the time limit.
- Fully heals tox and oxy damage.

Negatives:
- Consumes five times as much energy: must be paired with an upgraded powercell, or spare ones.
- Still cannot revive too damaged corpses (over 210 brute or burn)

Requires materials 6, bio 6, and power storage 5, costs gold and titanium on top of some metal and glass.

### CODE REVIEW

**Comment by Durkel** [comment]
Aren't autocloners enough?

**Comment by XDTM** [comment]
This is technically worse than autocloning

**Comment by GunHog** [comment]
What is the new time limit?

**Comment by XDTM** [comment]
Currently it's none

**Comment by Jalleo** [comment]
Needs a time limit but longer.

File: code/game/objects/items/weapons/defib.dm [hunk 1]
```diff
@@ -515,6 +529,23 @@
 				if(failed)
 					user.visible_message(failed)
 					playsound(get_turf(src), 'sound/machines/defib_failed.ogg', 50, 0)
+				else if (req_defib && defib.frankenstein) //If the defib is upgraded
```
Comment (JJRcop) [inline code/game/objects/items/weapons/defib.dm]: Whoaaaa... copypaste here.

**Comment by PKPenguin321** [comment]
Make the power drained a variable that increases with decay, so there's no hard limit on power drain, but if it's too far gone there simply won't be enough power in the defib to shock them

**Comment by GunHog** [comment]
That is a good idea, actually!

**Comment by kevinz000** [comment]
With power ratings at power cell power, 30
per second maybe. About 7-9 minutes max revive using an entire 15k cell

**Comment by Iamgoofball** [comment]
:-1:

use Strange Reagent to revive corpses past their prime

**Comment by Cheridan** [comment]
jesus christ please stop these """"PROGESSION""" memes where e verything in the fucking game is shit unless someone jumps through a R&D hoop at which time it gives you godmode

**Comment by Cheridan** [comment]
i mean god defibs need a buff but not like this, not "totally useless, until R&D gets their hands on them and then they replace cloning"

:rage4:

**Comment by AdamElTablawy** [comment]
how about removing fucking defibs and letting R&D print dildos so they can go **fuck themselves**

**Comment by Razharas** [comment]
^ salt

**Comment by Razharas** [comment]
No but seriously though upgrading should be gradual and not just broken mechanic-replacing monstruocity

**Comment by AdamElTablawy** [comment]
frankly i'm just annoyed with the current state of R&D altogether which is insert arbitrary items into the destructive analyzer to print things with ease that unlock arbitary functions that make machines less aids to use altogether until you pass through some stupid hoop to unlock godmode items that make you overpowered as hell

to the point where the solution to 'i coded something OP and fun' is 'let's lock it behind R&D with ridiculous levels'

it's just frankly not fun because you're not actually doing research, you're just following a spreadsheet to slap things in to get arbitrary shit, and the only way people managed to balance it was locking it behind yet more obnoxious shit such as glowshrooms and mining

to the point where absolutely any machine (see-autolathe, sleeper) is completely horrendous and mostly useless and obnoxious until you slap some parts into it thus making the majority of the game's not-hunting-antag fun (AKA being able to do your job without vomiting because of how slow everything is) depends on someone jamming their dick into the destructive analyzer while leaving other jobs to be neglected in favor of 'oh this job is lame lets give it buffs that are unlocked via R&D'

and now we come to the defibs, where you can apply some arbitrary patch to them to make them able to bypass it's entire balancing factor and make it completely superior to cloning because someone shat into the DA and got some upgrade for it

it's quite frankly just obnoxious and annoying, the entire system-both the stupid 'upgrades' which scale on a level of 'this is ass' to 'HOLY FUCK', the R&D system itself which is some spreadsheet you follow instead of actual research, and now we come to this abomination

tl;dr 

bad pr, defibs don't need this stupid 'wew we revive you instantly 1000000% better than cloning' buff, instead give science dildos so they can fuck themselves

**Comment by pudl** [comment]
Why does everything ever have to be upgradable? Cant you give it a rest?

**Comment by kevinz000** [comment]
haters gonna hate

**Comment by ExcessiveUseOfCobblestone** [comment]
I'm just not seeing how the disadvantages make a difference when if you can make this you probably have  a bs-batter[ies] and a fully maxed cell-charger. it makes the disadvantage nearly trivial except having to micromanage batteries.

**Comment by PKPenguin321** [comment]
it should always require more power than a regular defib no matter what

**Comment by optimumtact** [comment]
We don't need defibs and a cloner

**Comment by kevinz000** [comment]
NUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUUU
